File: core/src/main/java/net/dontdrinkandroot/wicket/component/basic/UnorderedList.java
package net.dontdrinkandroot.wicket.component.basic;

import java.util.List;

import org.apache.wicket.markup.ComponentTag;
import org.apache.wicket.model.IModel;


public abstract class UnorderedList<T> extends AbstractList<T> {

	public UnorderedList(String id, IModel<List<T>> model) {

		super(id, model);
	}


	@Override
	protected void onComponentTag(ComponentTag tag) {

		tag.setName("ul");
		super.onComponentTag(tag);
	}

}

File: core/src/main/kotlin/net/dontdrinkandroot/wicket/component/basic/UnorderedList.kt
package net.dontdrinkandroot.wicket.component.basic

import org.apache.wicket.Component
import org.apache.wicket.behavior.Behavior
import org.apache.wicket.markup.ComponentTag
import org.apache.wicket.model.IModel

abstract class UnorderedList<T>(
    id: String,
    model: IModel<List<T>>? = null,
    vararg behaviors: Behavior,
) : AbstractList<T>(id, model, *behaviors) {

    override fun onComponentTag(tag: ComponentTag) {
        tag.name = "ul"
        super.onComponentTag(tag)
    }
}

inline fun <T> unorderedList(
    id: String,
    model: IModel<List<T>>? = null,
    vararg behaviors: Behavior,
    crossinline createItemHandler: UnorderedList<T>.(id: String, model: IModel<T>) -> Component
) = object : UnorderedList<T>(id, model, *behaviors) {
    override fun createItem(id: String, model: IModel<T>) = createItemHandler(id, model)
}